java/server/src/org/openqa/selenium/grid/distributor/Distributor.java

```diff
@@ -160,8 +162,22 @@ protected Distributor(
           .with(new SpanDecorator(tracer, req -> "distributor.status")));
   }
 
-  public CreateSessionResponse newSession(HttpRequest request)
-      throws SessionNotCreatedException {
+  public CreateSessionResponse newSession(HttpRequest request) {
+    Either<SessionNotCreatedException, CreateSessionResponse> sessionResponse =
+      createNewSessionResponse(request);
+    if (sessionResponse.isRight()) {
+      return sessionResponse.right();
+    } else {
+      SessionNotCreatedException exception = sessionResponse.left();
+      if (exception instanceof RetrySessionRequestException) {
+        throw new SessionNotCreatedException(exception.getMessage(), exception);
+      }
+      throw sessionResponse.left();
+    }
+  }
+
+  public Either<SessionNotCreatedException, CreateSessionResponse> createNewSessionResponse(
+    HttpRequest request) throws SessionNotCreatedException {
 
     Span span = newSpanAsChildOf(tracer, request, "distributor.new_session");
```

Comment:
Shouldn't this span reflect the method where it is placed?

Reply:
+1